include/nighthawk/common/rate_limiter.h

```diff
@@ -24,6 +28,12 @@ class RateLimiter {
    * Releases a controlled resource.
    */
   virtual void releaseOne() PURE;
+
+  /**
+   * @return Envoy::TimeSource& time_source
+   */
+  virtual Envoy::TimeSource& timeSource() PURE;
+  virtual std::chrono::nanoseconds elapsed() PURE;
```

Comment:
Can we add more docs here? Like, what does `elapsed` do and what is it good for?

Reply:
Done in 4d6448b1e80306c0c20ef3dc1c63ba16fec5f2ee